lib/facter/resolvers/linux/networking.rb

```diff
@@ -47,6 +48,14 @@ def interfaces_mtu_and_index
             mtu_and_indexes
           end
 
+          def physical(interface_name)
+            if File.exist?("/sys/class/net/#{interface_name}/device")
+              @fact_list[:interfaces][interface_name][:physical] = true
+            else
+              @fact_list[:interfaces][interface_name][:physical] = false
+            end
+          end
+
```

Comment:
Could you add a test for this?

First update the expected result to include `physical: true` to https://github.com/puppetlabs/facter/blob/83ac46d0e6f9b8fc525cde8388ab5d0695d99ca8/spec/facter/resolvers/linux/networking_spec.rb#L262-L287

Next mock the behavior, adding something like the follow to [here](https://github.com/puppetlabs/facter/blob/83ac46d0e6f9b8fc525cde8388ab5d0695d99ca8/spec/facter/resolvers/linux/networking_spec.rb#L295)

```ruby
allow(File).to receive(:exist?).with("/sys/class/net/lo/device").and_return(true)
allow(File).to receive(:exist?).with("/sys/class/net/ens160/device").and_return(true)
```

And then the expectation will automatically be verified in https://github.com/puppetlabs/facter/blob/83ac46d0e6f9b8fc525cde8388ab5d0695d99ca8/spec/facter/resolvers/linux/networking_spec.rb#L301

Reply:
I took a stab at it, is that better?